File: Python/baekjoon/단계별로 풀어보기/기본 수학2/1978_소수 찾기.py
# https://www.acmicpc.net/problem/1978

import math

input()
nums = list(map(int, input().split()))
answer = 0
for num in nums:
    if num == 1:
        continue
    for i in range(2, math.ceil(math.sqrt(num)) + 1):
        if num % i == 0:
            break
    if (num % i != 0) or (num == 2):
        answer += 1
print(answer)


# def prime(n):
#     import math
#     if n < 2:
#         return False
#     elif n == 2:
#         return True
#     else:
#         for i in range(2, math.ceil(math.sqrt(n)) + 1):
#             if n % i == 0:
#                 return False
#         return True
#
#
# input()
# nums = list(map(int, input().split()))
# answer = 0
# for num in nums:
#     answer += prime(num)
# print(answer)

File: C#/Baekjoon/약수, 배수와 소수/1978_소수 찾기.cs
//https://www.acmicpc.net/problem/1978

bool[] p = new bool[1001];
p[0] = true;
p[1] = true;
for (int i = 2; i < 1001; i++)
{
    if (p[i] == false)
    {
        for (int j = i; j * i < 1001; j++)
        {
            p[j * i] = true;
        }
    }
}
Console.ReadLine();
int[] n = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
int cnt = 0;
foreach (int i in n)
{
    if (p[i] == false) cnt++;
}
Console.Write(cnt);